src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/DetectPreviewFeatureAnalyzer.cs

```diff
@@ -225,33 +240,39 @@ public override void Initialize(AnalysisContext context)
             return null;
         }
 
-        private void ProcessFieldSymbolAttributes(SymbolAnalysisContext context, IFieldSymbol symbol,
-            ConcurrentDictionary<ISymbol, bool> requiresPreviewFeaturesSymbols,
-            INamedTypeSymbol previewFeatureAttributeSymbol)
+        private void ProcessFieldSymbolAttributes(SymbolAnalysisContext context,
+                                                  IFieldSymbol symbol,
+                                                  ConcurrentDictionary<ISymbol, bool> requiresPreviewFeaturesSymbols,
+                                                  ConcurrentDictionary<ISymbol, ValueTuple<string?, string?>> previewSymbolsToMessageAndUrl,
+                                                  INamedTypeSymbol previewFeatureAttributeSymbol)
         {
             ISymbol symbolType = symbol.Type;
             while (symbolType is IArrayTypeSymbol arrayType)
             {
                 symbolType = arrayType.ElementType;
             }
 
-            ProcessFieldOrEventSymbolAttributes(context, symbol, symbolType, requiresPreviewFeaturesSymbols, previewFeatureAttributeSymbol);
+            ProcessFieldOrEventSymbolAttributes(context, symbol, symbolType, requiresPreviewFeaturesSymbols, previewSymbolsToMessageAndUrl, previewFeatureAttributeSymbol);
         }
 
-        private void ProcessFieldOrEventSymbolAttributes(SymbolAnalysisContext context, ISymbol symbol, ISymbol symbolType,
-            ConcurrentDictionary<ISymbol, bool> requiresPreviewFeaturesSymbols,
-            INamedTypeSymbol previewFeatureAttributeSymbol)
+        private void ProcessFieldOrEventSymbolAttributes(SymbolAnalysisContext context,
+                                                         ISymbol symbol,
+                                                         ISymbol symbolType,
+                                                         ConcurrentDictionary<ISymbol, bool> requiresPreviewFeaturesSymbols,
+                                                         ConcurrentDictionary<ISymbol, ValueTuple<string?, string?>> previewSymbolsToMessageAndUrl,
+                                                         INamedTypeSymbol previewFeatureAttributeSymbol)
         {
             if (SymbolIsAnnotatedAsPreview(symbolType, requiresPreviewFeaturesSymbols, previewFeatureAttributeSymbol))
             {
+                string message = string.Format(CultureInfo.CurrentCulture, (string)s_fieldOrEventIsPreviewTypeMessage, symbol.Name, symbolType.Name);
```

Comment:
1 example of where I'm explicitly converting LocalizableString -> string.

Reply:
Hm, once i had a comment do not use manual string concatenation, each case should have its own resource string:
Is the resource string versions above could work for this:

- When no message: `'{0}' implements the preview interface '{1}' and therefore needs to opt into preview features. See {2} for more information.`
- When there is a message: `{3} '{0}' implements the preview interface '{1}' and therefore needs to opt into preview features. See {2} for more information.`

Reply:
Or we could just simply use for both `{3} '{0}' implements the preview interface '{1}' and therefore needs to opt into preview features. See {2} for more information.` so that when there is no message there will be one space before the message

Reply:
Cool, I ended up doing the 2nd suggestion.